engine/CMakeLists.txt

```diff
@@ -33,6 +33,10 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.16 FATAL_ERROR)
 
+IF (POLICY CMP0167)
+    CMAKE_POLICY (SET CMP0167 OLD)
```

Comment:
Even with the Setup Python action in place, the Mac builds still didn't work until I added this to CMakeLists.txt

Reply:
We're trailing behind CMake here. I'm not wild about that, as we're creating technical debt. However, I don't see another way, as we're resource constrained.

Reply:
@royfalk agreed. Also note that in order to get rid of this bit of technical debt, I think we'll have to remove support for CMake versions 3.29 and earlier.